packages/git-diff/lib/main.js

```diff
@@ -1,32 +1,52 @@
-const GitDiffView = require('./git-diff-view');
-const DiffListView = require('./diff-list-view');
+'use babel';
+
+import { CompositeDisposable } from 'atom';
+import GitDiffView from './git-diff-view';
+import DiffListView from './diff-list-view';
 
 let diffListView = null;
+let diffViews = null;
+let subscriptions = null;
+
+export default {
+  activate(state) {
+    subscriptions = new CompositeDisposable();
+    diffViews = new Set();
+
+    subscriptions.add(
+      atom.workspace.observeTextEditors(editor => {
+        const editorElm = atom.views.getView(editor);
+        const diffView = new GitDiffView(editor);
 
-module.exports = {
-  activate() {
-    const watchedEditors = new WeakSet();
-
-    atom.workspace.observeTextEditors(editor => {
-      if (watchedEditors.has(editor)) return;
-
-      new GitDiffView(editor).start();
-      atom.commands.add(
-        atom.views.getView(editor),
-        'git-diff:toggle-diff-list',
-        () => {
-          if (diffListView == null) diffListView = new DiffListView();
-          diffListView.toggle();
-        }
-      );
-
-      watchedEditors.add(editor);
-      editor.onDidDestroy(() => watchedEditors.delete(editor));
-    });
+        diffViews.add(diffView);
+
+        let editorSubs;
+        const command = 'git-diff:toggle-diff-list';
+        subscriptions.add(
+          (editorSubs = new CompositeDisposable(
+            atom.commands.add(editorElm, command, () => {
+              if (diffListView == null) diffListView = new DiffListView();
+              diffListView.toggle();
+            }),
+            editor.onDidDestroy(() => {
+              diffView.destroy();
+              diffViews.delete(diffView);
+              editorSubs.dispose();
+              subscriptions.remove(editorSubs);
+            })
+          ))
+        );
+      })
+    );
   },
 
   deactivate() {
-    if (diffListView) diffListView.destroy();
     diffListView = null;
+
+    for (const v of diffViews) v.destroy();
```

Comment:
```suggestion
    for (const diffView of diffViews) diffView.destroy();
```

Reply:
That one I suppose I can *"fix"* too, although I don't see the problem since It's only a single line and it's very clear we're itterating through diffViews.

Reply:
You'd better choose the words that are searchable by find and replace.  I suggestion this one:

```suggestion
    for (const diffView of diffViews) { 
		diffView.destroy();
	}
```

The above suggestion is debuggable (a breakpoint can be added), and it is also searchable

Reply:
Your wording's a little confusing. I got the debugging bit from another comment, but what do you mean by 
> You'd better choose the words that are searchable by find and replace
Just because an itterable's variable name is a plural noun, doesn't mean all variables accessing it are guaranteed to use the same singular noun. Although that would be helpful for consistency.

Maybe I just answered my own question?